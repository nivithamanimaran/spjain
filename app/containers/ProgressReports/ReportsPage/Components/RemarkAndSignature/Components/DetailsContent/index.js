import React from 'react';
import { makeStyles, Box, Typography, Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';

const useStyles = makeStyles(theme => ({
  container: {
    height: '100%',
  },
  subHeadings: {
    color: theme.palette.KenColors.neutral61,
    fontSize: '13px',
  },
  text: {
    color: theme.palette.KenColors.neutral103,
    fontSize: '13px',
    fontWeight: 'bold',
  },
  remarkText: {
    color: theme.palette.KenColors.neutral61,
    fontSize: '13px',
  },
  box: {
    backgroundColor: theme.palette.KenColors.tertiaryBlue40,
    padding: '16px',
    borderRadius: '6px',
    // minHeight: '100vh',
    height: '100%',
  },
  headings: {
    color: theme.palette.KenColors.tertiaryBlue502,
    fontSize: '12px',
    paddingBottom: '16px',
  },
}));

const DetailsContent = props => {
  const { title, data, variant } = props;
  const { t } = useTranslation();
  const classes = useStyles();

  return (
    <Box className={classes.container} data-testid="details-content">
      <Box className={classes.box}>
        <Typography
          className={classes.headings}
          data-testid="details-content-title"
        >
          {title}
        </Typography>
        <Grid container spacing={4} direction="column">
          <>
            {data &&
              Array.isArray(data) &&
              data.map((item, index) => {
                return (
                  <Grid item xs={12} key={index}>
                    <Typography
                      data-testid="student-details-label"
                      className={classes.subHeadings}
                      color="primary"
                    >
                      {item.label}
                    </Typography>
                    <Typography
                      data-testid="student-details-value"
                      className={classes.text}
                      variant="subtitle1"
                      color="primary"
                    >
                      {item.value}
                    </Typography>
                  </Grid>
                );
              })}
          </>
        </Grid>
      </Box>
    </Box>
  );
};
export default DetailsContent;
