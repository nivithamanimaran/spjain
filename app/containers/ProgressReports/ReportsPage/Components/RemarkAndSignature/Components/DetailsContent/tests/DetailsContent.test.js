import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import DetailsContent from '../index';
import ThemeProvider from '../../../../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

describe('Details Content component', () => {
  const div = document.createElement('div');
  const details = [{ label: 'Height', value: '109cm' }];
  it('Details Content component renders correctly without props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <DetailsContent />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <DetailsContent />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('details-content')).toMatchSnapshot();
  });
  it('Details Content component renders correctly with props', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <DetailsContent />
        </Router>
      </ThemeProvider>,
      div
    );

    const { queryByTestId } = render(
      <ThemeProvider>
        <Router>
          <DetailsContent title={'title'} data={details} />
        </Router>
      </ThemeProvider>
    );

    const title = queryByTestId('details-content-title');
    expect(title).toBeTruthy();
    expect(title.textContent).toBe('title');

    const label = queryByTestId('student-details-label');
    expect(label).toBeTruthy();
    expect(label.textContent).toBe('Height');

    const value = queryByTestId('student-details-value');
    expect(value).toBeTruthy();
    expect(value.textContent).toBe('109cm');
  });
});
