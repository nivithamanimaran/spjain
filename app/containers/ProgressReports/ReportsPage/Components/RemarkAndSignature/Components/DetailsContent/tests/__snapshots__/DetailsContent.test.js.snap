// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Content component Details Content component renders correctly without props 1`] = `
<div
  class="MuiBox-root MuiBox-root-9 makeStyles-container-1"
  data-testid="details-content"
>
  <div
    class="MuiBox-root MuiBox-root-10 makeStyles-box-5"
  >
    <p
      class="MuiTypography-root makeStyles-headings-6 MuiTypography-body1"
      data-testid="details-content-title"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-direction-xs-column"
    />
  </div>
</div>
`;
